app/controllers/invoices_controller.rb

```diff
@@ -1,7 +1,144 @@
 class InvoicesController < ApplicationController
 
+  before_action :authenticate_user!
+  load_and_authorize_resource
+
+  def index
+    @invoices = current_company.invoices
+  end
+
   def new
     @invoice = Invoice.new
     @invoice.line_items << LineItem.new
   end
+
+  def create
+    @invoice = Invoice.new(invoice_params)
+    @invoice.address = Address.new.tap do |a|
+      invoice_address = @invoice.client.address
+      a.street_1 = invoice_address.street_1
+      a.street_2 = invoice_address.street_2
+      a.city = invoice_address.city
+      a.state = invoice_address.state
+      a.pincode = invoice_address.pincode
+      a.country_code = invoice_address.country_code
+    end
+    if params[:email_send] == "true"
+      @invoice.status = 'sent'
+    else
+      @invoice.status = 'draft'
+    end
+    if @invoice.save
+      if params[:email_send] == "true"
+        pdf_render
+        ClientMailer.send_email(@invoice.client,@kit.to_pdf, @invoice, current_user).deliver
```

Comment:
No need to pass `@invoice.client` as you already passing `@invoice` as third param